specification.markdown

```diff
@@ -476,6 +476,18 @@ A purpose attribute distinguishes the meaning of the documentation. Values for t
 
 The purpose attribute is REQUIRED.
 
+### \<implementation_note> Element
+
+In some cases within an EBML Document Type, the attributes of the `<element>` element are not sufficient to clearly communicate how the defined EBML Element is intended to be implemented. For instance, one EBML Element might only be mandatory if another EBML Element is present, or as another example, the default value of an EBML Element might derive from a related Element's content. In these cases where the Element's definition is conditional or advanced implementation notes are needed, one or many `<implementation_note>` elements can be used to store that information. The `<implementation_note>` refer to a specific attribute of the parent `<element>` as expressed by the `note_type` attribute (#note_type).
+
+### \<implementation_note> Attributes
+
+#### note_type
+
+The note_type attribute references which of the `<element>`'s attributes that the implementation_note is in regards to. The note_type attribute MUST be set to one of the following values (corresponding to that attribute of the parent `<element>`): `minOccurs`, `maxOccurs`, `range`, `length`, or `default`. The `<implementation_note>` SHALL supersede the parent `<element>`'s attribute that is named in the `note_type` attribute. An `<element>` SHALL NOT have more than one `<implementation_note>` of the same `note_type`.
```

Comment:
no 'minver' / 'maxver' ? For DivX we MAY explain why elements not found in Matroska can actually be found in Matroska v2 files.

Reply:
Good point. Will add.

Reply:
added